### PR TITLE
Transition to pyproject.toml and setup.cfg

### LINKED ISSUE #301: Transition Compose to use pyproject.toml and setup.cfg (moving away from setup.py)
> The future of Python packaging is pyproject.toml, and (for now) setup.cfg, based on [PEP 518](https://snarky.ca/clarifying-pep-518/) and (soon) [PEP 621](https://www.python.org/dev/peps/pep-0621/).

- http://ivory.idyll.org/blog/2021-transition-to-pyproject.toml-example.html
- Therefore we should migrate our Woodwork package to follow this new standard
- Examples: 
- https://github.com/alteryx/woodwork/issues/1286
- https://github.com/alteryx/featuretools/pull/1941

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #301

### CODE REVIEW

File: .github/workflows/latest_dependency_checker.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         source venv_core/bin/activate
         python -m pip install --upgrade pip
         python -m pip install -e .
```
Comment (thehomebrewnerd) [inline .github/workflows/latest_dependency_checker.yml:27]: I think this line could be removed as `python -m pip install .[test]` should also install the base library and its requirements in addition to the test requirements.

File: .github/workflows/lint_check.yml [hunk 2]
```diff
@@ -7,30 +7,29 @@ on:
...
+      - name: Install compose with dev, and test requirements
         run: |
           pip config --site set global.progress_bar off
           python -m pip install --upgrade pip
-          python -m pip install dist/package/
-          python -m pip install -r dev-requirements.txt
-          python -m pip install -r test-requirements.txt
-      - name: Run lint tests
-        run: cd dist/package && make lint -f ../../Makefile
+          python -m pip install -e .[dev]
```
Comment (thehomebrewnerd) [inline .github/workflows/lint_check.yml:33]: This probably doesn't need to be an editable install.

File: .github/workflows/unit_tests_with_latest_deps.yml [hunk 3]
```diff
@@ -24,19 +24,24 @@ jobs:
...
       - name: Install package with test requirements
         run: |
           python -m pip config --site set global.progress_bar off
           python -m pip install --upgrade pip
-          python -m pip install dist/package/
-          python -m pip install -r dist/package/test-requirements.txt
-      - if: ${{ matrix.python-version != 3.8 }}
-        name: Run unit tests (no code coverage)
-        run: cd dist/package && make test -f ../../Makefile
+          python -m pip install -e unpacked_sdist/
```
Comment (thehomebrewnerd) [inline .github/workflows/unit_tests_with_latest_deps.yml:32]: I don't think this line is necessary - the line below should cover it. I think. Also I don't think these need to be editable installs.

File: Makefile [hunk 4]
```diff
@@ -1,42 +1,50 @@
...
 	python -m pip install --upgrade pip
+
+.PHONY: upgradebuild
+upgradebuild:
+	python -m pip install --upgrade build
+
+.PHONY: package_compose
+package_compose: upgradepip upgradebuild
+	python -m build
+	$(eval FT_VERSION := $(shell grep '__version__\s=' composeml/version.py | grep -o '[^ ]*$$'))
```
Comment (thehomebrewnerd) [inline Makefile:48]: Minor detail, but should `FT_VERSION` be `COMPOSE_VERSION` or just `VERSION`?

**Review by thehomebrewnerd — DISMISSED** [review]
Looks pretty good - left a few comments on some minor things if you want to update, but nothing that is critical to fix.

File: Makefile (hunk 4 above)
Comment (gsheni) [inline Makefile:48]:
```suggestion
	$(eval COMPOSE_VERSION := $(shell grep '__version__\s=' composeml/version.py | grep -o '[^ ]*$$'))
```

File: Makefile [hunk 5]
```diff
@@ -1,42 +1,50 @@
...
+
+.PHONY: upgradebuild
+upgradebuild:
+	python -m pip install --upgrade build
+
+.PHONY: package_compose
+package_compose: upgradepip upgradebuild
+	python -m build
+	$(eval COMPOSE_VERSION := $(shell grep '__version__\s=' composeml/version.py | grep -o '[^ ]*$$'))
+	tar -zxvf "dist/composeml-${FT_VERSION}.tar.gz"
```
Comment (gsheni) [inline Makefile:49]:
```suggestion
	tar -zxvf "dist/composeml-${COMPOSE_VERSION}.tar.gz"
```

File: Makefile [hunk 6]
```diff
@@ -1,42 +1,50 @@
...
+.PHONY: upgradebuild
+upgradebuild:
+	python -m pip install --upgrade build
+
+.PHONY: package_compose
+package_compose: upgradepip upgradebuild
+	python -m build
+	$(eval COMPOSE_VERSION := $(shell grep '__version__\s=' composeml/version.py | grep -o '[^ ]*$$'))
+	tar -zxvf "dist/composeml-${COMPOSE_VERSION}.tar.gz"
+	mv "composeml-${FT_VERSION}" unpacked_sdist
```
Comment (gsheni) [inline Makefile:50]:
```suggestion
	mv "composeml-${COMPOSE_VERSION}" unpacked_sdist
```

File: .github/workflows/latest_dependency_checker.yml (hunk 1 above)
Comment (gsheni) [inline .github/workflows/latest_dependency_checker.yml:27]:
```suggestion
```

File: .github/workflows/lint_check.yml (hunk 2 above)
Comment (gsheni) [inline .github/workflows/lint_check.yml:33]:
```suggestion
          python -m pip install .[dev]
```

File: .github/workflows/unit_tests_with_latest_deps.yml (hunk 3 above)
Comment (gsheni) [inline .github/workflows/unit_tests_with_latest_deps.yml:32]:
```suggestion
```

File: .github/workflows/unit_tests_with_latest_deps.yml [hunk 7]
```diff
@@ -24,19 +24,23 @@ jobs:
...
       - name: Install package with test requirements
         run: |
           python -m pip config --site set global.progress_bar off
           python -m pip install --upgrade pip
-          python -m pip install dist/package/
-          python -m pip install -r dist/package/test-requirements.txt
-      - if: ${{ matrix.python-version != 3.8 }}
-        name: Run unit tests (no code coverage)
-        run: cd dist/package && make test -f ../../Makefile
+          python -m pip install -e unpacked_sdist/[test]
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_latest_deps.yml:32]:
```suggestion
          python -m pip install unpacked_sdist/[test]
```

**Review by thehomebrewnerd — APPROVED** [review]

**Pull request merged**